src/libraries/System.Private.CoreLib/src/System/MissingMemberException.cs

```diff
@@ -71,5 +74,184 @@ public override string Message
+                    break;
+                case CorElementType.ELEMENT_TYPE_R4:
+                    stringBuilder.Append("float");
+                    break;
+                case CorElementType.ELEMENT_TYPE_R8:
+                    stringBuilder.Append("double");
+                    break;
+                case CorElementType.ELEMENT_TYPE_STRING:
+                    stringBuilder.Append("String");
+                    break;
+                case CorElementType.ELEMENT_TYPE_VAR:
+                case CorElementType.ELEMENT_TYPE_OBJECT:
+                    stringBuilder.Append("Object");
+                    break;
+
+                case CorElementType.ELEMENT_TYPE_PTR:
+                    bytesConsumed += UnparseType(sig.Slice(1), ref stringBuilder);
+                    stringBuilder.Append('*');
+                    break;
+
+                case CorElementType.ELEMENT_TYPE_BYREF:
+                    bytesConsumed += UnparseType(sig.Slice(1), ref stringBuilder);
+                    stringBuilder.Append('&');
+                    break;
+
+                case CorElementType.ELEMENT_TYPE_VALUETYPE:
+                case CorElementType.ELEMENT_TYPE_CLASS:
+                    int length = sig.Slice(1).IndexOf((byte)'\0');
+                    if (length == -1)
+                    {
+                        throw new ArgumentOutOfRangeException(); // rethrown by caller
+                    }
+                    stringBuilder.Append(Encoding.UTF8.GetString(sig.Slice(1, length)));
+                    bytesConsumed += length + 1;
+                    break;
+
+                case CorElementType.ELEMENT_TYPE_SZARRAY:
+                    bytesConsumed += UnparseType(sig.Slice(1), ref stringBuilder);
+                    stringBuilder.Append("[]");
+                    break;
+
+                case CorElementType.ELEMENT_TYPE_ARRAY:
+                    bytesConsumed += UnparseType(sig.Slice(1), ref stringBuilder);
+                    sig = sig.Slice(1 + bytesConsumed);
+                    int rank = BinaryPrimitives.ReadInt32LittleEndian(sig);
+                    sig = sig.Slice(sizeof(int));
+                    bytesConsumed += sizeof(int);
+                    if (rank > 0)
+                    {
+                        int nSizes = BinaryPrimitives.ReadInt32LittleEndian(sig);
+                        sig = sig.Slice(sizeof(int) * (nSizes + 1));
+                        bytesConsumed += sizeof(int) * (nSizes + 1);
+
+                        int nBounds = BinaryPrimitives.ReadInt32LittleEndian(sig);
+                        // sig = sig.Slice(sizeof(int) * (nBounds + 1));
+                        bytesConsumed += sizeof(int) * (nBounds + 1);
+
+                        stringBuilder.Append('[');
+                        for (int i = 1; i < rank; i++)
+                        {
+                            stringBuilder.Append(',');
+                        }
+                        stringBuilder.Append(']');
```

Comment:
The native code formats this into jagged array. Does this mean multi-dimensional array?
Malformed rank may cause OOM here. Should we add a limitation?